common/djangoapps/entitlements/models.py

```diff
@@ -162,6 +162,7 @@ class CourseEntitlement(TimeStampedModel):
         blank=True
     )
     order_number = models.CharField(max_length=128, null=True)
+    marked_unrefundable = models.BooleanField(default=False)
```

Comment:
I am not sure I like this name here.  I think this is a good route, but we should consider something that highlights the goal a little more.

Maybe refund_locked, refund_blocked, manual_refund_lock, etc.  What are your thoughts?